### PR TITLE
Set entity world early

### PR DESCRIPTION
**Pull request by mathiascode**
In hindsight, I'm not happy with https://github.com/cuberite/cuberite/pull/4606. It seems too dangerous, and plugins probably have to possibility to crash the server while the world is unset.

Closes https://github.com/cuberite/cuberite/issues/4703

### CODE REVIEW

**Comment by tigerw** [comment]
looks good

File: src/World.cpp [hunk 1]
```diff
@@ -1024,7 +1024,6 @@ void cWorld::Tick(std::chrono::milliseconds a_Dt, std::chrono::milliseconds a_La
 	}
 	for (auto & Entity : EntitiesToAdd)
 	{
-		Entity->SetWorld(this);
```
Comment (peterbell10) [inline src/World.cpp:1027]: Can you assert the world is set correctly.

**Comment by mathiascode** [comment]
Did not trigger in my testing.
@Seadragon91 Could you confirm that this fixes your issue?

**Comment by Seadragon91** [comment]
Did a few other tests, too. Had a few deadlocks during testing, but they are already known.

**Pull request merged**